src/OpenAI/Model/Gpt/Version.php

```diff
@@ -4,19 +4,57 @@
 
 namespace PhpLlm\LlmChain\OpenAI\Model\Gpt;
 
-enum Version: string
+use Webmozart\Assert\Assert;
+
+final class Version
 {
-    case GPT_35_TURBO = 'gpt-3.5-turbo';
-    case GPT_35_TURBO_INSTRUCT = 'gpt-3.5-turbo-instruct';
-    case GPT_4 = 'gpt-4';
-    case GPT_4_TURBO = 'gpt-4-turbo';
-    case GPT_4o = 'gpt-4o';
-    case GPT_4o_MINI = 'gpt-4o-mini';
-    case o1_MINI = 'o1-mini';
-    case o1_PREVIEW = 'o1-preview';
-
-    public function supportsStructuredOutput(): bool
-    {
-        return self::GPT_4o === $this || self::GPT_4o_MINI === $this;
+    /**
+     * @param non-empty-string $name
+     */
+    public function __construct(
+        public string $name,
+        public bool $supportStructuredOutput,
```

Comment:
can we add `readonly` or do i miss sth?

```suggestion
        public readonly string $name,
        public readonly bool $supportStructuredOutput,
```

Reply:
basically same questions for the other ones.

i guess on class level it's not possible due to named constructors, right?

Reply:
done